tests/helpers/main.cu

```diff
@@ -1,27 +1,60 @@
 #include <cuda_runtime.h>
 #include <cstdio>
+#include <map>
 
 #include "flamegpu/gpu/CUDASimulation.h"
 #include "gtest/gtest.h"
 #include "helpers/device_initialisation.h"
+#include "flamegpu/util/Environment.h"
+#include "flamegpu/io/Telemetry.h"
 
 
 GTEST_API_ int main(int argc, char **argv) {
-  // Time the cuda agent model initialisation, to check it creates the context.
-  flamegpu::tests::timeCUDASimulationContextCreationTest();
-  // Run the main google test body
-  printf("Running main() from %s\n", __FILE__);
-  testing::InitGoogleTest(&argc, argv);
-  auto rtn = RUN_ALL_TESTS();
-  // Reset all cuda devices for memcheck / profiling purposes.
-  int devices = 0;
-  gpuErrchk(cudaGetDeviceCount(&devices));
-  if (devices > 0) {
-    for (int device = 0; device < devices; ++device) {
-      gpuErrchk(cudaSetDevice(device));
-      gpuErrchk(cudaDeviceReset());
+    // Telemetry
+    const bool share_usage = flamegpu::io::Telemetry::globalTelemetryEnabled();
+    flamegpu::util::setEnvironmentVariable("FLAMEGPU_SHARE_USAGE_STATISTICS", "False");    // Disable any global telemetry to prevent telemetry for every simulation in test suite
+    flamegpu::io::Telemetry::silenceTelemetryNotice();                                     // Silence telemetry notice warning in test suite
+    // Time the cuda agent model initialisation, to check it creates the context.
+    flamegpu::tests::timeCUDASimulationContextCreationTest();
+    // google test doesnt have a verbose mode so check for verbosity flag for telemetry output
+    bool verbose = false;
+    for (int i = 0; i < argc; i++) {
+        std::string arg = std::string(argv[i]);
+        if ((arg.compare("-v") == 0) || (arg.compare("--verbose") == 0))
+            verbose = true;
+    }
+    // Run the main google test body
+    printf("Running main() from %s\n", __FILE__);
+    testing::InitGoogleTest(&argc, argv);
+    auto rtn = RUN_ALL_TESTS();
+    // send telemetry if it was enabled globally (dont bother nagging on test suite)
+    if (share_usage) {
```

Comment:
I don't see the point in having the test suite phone home about how successful it was or not. It's meaningless without knowing if the test suite has been mutated or not.

Reply:
It will also be phoning home for each individual test when `USE_GTEST_DISCOVER` is on. Perhaps only phone home when more than a single test is being ran to prevent 1000 telementry requests per ctest run?

Reply:
Only running if more than 1 test is to be exeucted.